File: demo_app/app/src/main/java/com/bandyer/demo_android_sdk/adapter_items/UserItem.java
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_android_sdk.adapter_items;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.bandyer.demo_android_sdk.R;
import com.mikepenz.fastadapter.items.AbstractItem;

import java.util.List;

import butterknife.BindView;

/**
 * A simple RecyclerView item used to display the user alias as a cell in the list.
 */
public class UserItem extends AbstractItem<UserItem, UserItem.ViewHolder> {

    public final String userAlias;

    public UserItem(String userAlias) {
        this.userAlias = userAlias;
    }

    @Override
    public int getType() {
        return R.id.user_item_id;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.item_user;
    }

    @NonNull
    @Override
    public ViewHolder getViewHolder(@NonNull View v) {
        return new ViewHolder(v);
    }

    protected static class ViewHolder extends ButterKnifeViewHolder<UserItem> {

        @BindView(R.id.userAlias)
        TextView userAlias;

        ViewHolder(View view) {
            super(view);
        }

        @Override
        public void bindView(@NonNull UserItem item, @NonNull List<Object> payloads) {
            userAlias.setText(item.userAlias);
        }

        @Override
        public void unbindView(@NonNull UserItem item) {
            userAlias.setText(null);
        }
    }
}

File: app_utilities/src/main/java/com/bandyer/app_utilities/adapter_items/UserItem.kt
/*
 * Copyright (C) 2020 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */
package com.bandyer.app_utilities.adapter_items

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bandyer.app_utilities.R
import com.mikepenz.fastadapter.items.AbstractItem
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_user_layout.view.*

/**
 * A simple RecyclerView item used to display the user userAlias as a cell in the list.
 */
class UserItem(@JvmField val userAlias: String) : AbstractItem<UserItem.ViewHolder>() {

    override var identifier: Long = userAlias.hashCode().toLong()
    override val type: Int = R.id.user_item_id
    override val layoutRes: Int = R.layout.item_user_layout
    override fun getViewHolder(v: View): ViewHolder = ViewHolder(v)

    override fun bindView(holder: ViewHolder, payloads: List<Any>) {
        super.bindView(holder, payloads)
        holder.containerView.userAlias!!.text = userAlias
    }

    override fun unbindView(holder: ViewHolder) {
        super.unbindView(holder)
        holder.containerView.userAlias!!.text = null
    }

    class ViewHolder(val containerView: View) : RecyclerView.ViewHolder(containerView)
}
